### ISSUE TITLE
Rule: no-space-before-property

### ISSUE THREAD

**Issue by puzrin** [eslint/eslint#1086]
If expression is inline, whitespace should not be allowed after dot or before square brackets.

valid:

```
foo[bar]

foo
  .bar()
  .baz()

foo.
  bar().
  baz()
```

invalid:

```
foo [bar]

foo. bar. baz
```

**Comment by gyandeeps** [comment]
@nzakas Based on the comments I think we are missing the `enhancement` label here. right?

**Comment by nzakas** [comment]
Possible overlaps #746

**Comment by scinos** [comment]
Any plans to merge this in ESLint?

**Comment by nzakas** [comment]
@scinos we are looking for someone to volunteer. Interested?

**Comment by kaicataldo** [comment]
I'd like to take a crack at this this week (have another issue in the pipeline as well)

**Comment by nzakas** [comment]
Awesome, thank you

**Comment by kaicataldo** [comment]
Ended up being busier than I anticipated, but I'm still interested in doing this. I'm not entirely sure what my schedule is going to look like with the upcoming holidays, so if someone else wants to try their hand please feel free and leave a note here. Don't want to hold anything up :)

**Comment by nzakas** [comment]
Thanks @kaicataldo

**Comment by mysticatea** [comment]
BTW, is there a rule for spaces followed by a dot?

```js
foo .bar;
```

If no, I prefer a rule like `dot-spacing` instead.
The rule is for spaces before and after a dot of MemberExpression.

**Comment by nzakas** [comment]
@mysticatea I believe that is what is being proposed here.

**Comment by mysticatea** [comment]
Oh sorry. nevermind.

**Comment by kaicataldo** [comment]
Starting on this tonight
